PROTOCOL.md

```diff
@@ -1343,6 +1344,18 @@ When enabled:
  - The value of `delta.generationExpression` SHOULD be parsed as a SQL expression.
  - Writers MUST enforce that any data writing to the table satisfy the condition `(<value> <=> <generation expression>) IS TRUE`. `<=>` is the NULL-safe equal operator which performs an equality comparison like the `=` operator but returns `TRUE` rather than NULL if both operands are `NULL`
 
+## Default Columns
+
+Delta supports defining default expressions for columns on Delta tables. Delta will generate default values for columns when users do not explicitly provide values for them when writing to such tables, or when the user explicitly specifies the `DEFAULT` SQL keyword for any such column.
+ 
+Enablement:
+ - The table must be on Writer Version 7, and a feature name `allowColumnDefaults` must exist in the table `protocol`'s `writerFeatures`.
+
+When enabled:
+ - The `metadata` for the column in the table schema MAY contain the key `CURRENT_DEFAULT`.
+ - The value of `CURRENT_DEFAULT` SHOULD be parsed as a SQL expression. Any engine that assigns this value can use its own SQL dialect of choice to represent the expression as a string, and use that same dialect to evaluate that expression later for future writes. If one engine writes the string metadata using its own SQL dialect and another engine then consumes it later when performing writes, the results are undefined.
```

Comment:
> Any engine that assigns this value can use its own SQL dialect of choice to represent the expression as a string, and use that same dialect to evaluate that expression later for future writes. If one engine writes the string metadata using its own SQL dialect and another engine then consumes it later when performing writes, the results are undefined.

For other features, like invariant and check constraint, we only say SQL expression. I assume it would be standard SQL. Accepting any SQL dialect seems dangerous to me as it would make the table dependent on a specific engine.

Reply:
OK, I updated this to just say "SQL expression" like we do for the "Generated Columns" section above.